server/forge/forgejo/forgejo.go

```diff
@@ -643,34 +667,75 @@ func (c *Forgejo) getChangedFilesForPR(ctx context.Context, repo *model.Repo, in
 	}, -1)
 }
 
-func (c *Forgejo) getTagCommitSHA(ctx context.Context, repo *model.Repo, tagName string) (string, error) {
+func (c *Forgejo) getTagMessage(ctx context.Context, repo *model.Repo, tagName string) (*model.Commit, string, error) {
 	_store, ok := store.TryFromContext(ctx)
 	if !ok {
-		log.Error().Msg("could not get store from context")
-		return "", nil
+		return nil, "", fmt.Errorf("could not get store from context")
 	}
 
 	repo, err := _store.GetRepoNameFallback(repo.ForgeRemoteID, repo.FullName)
 	if err != nil {
-		return "", err
+		return nil, "", err
 	}
 
 	user, err := _store.GetUser(repo.UserID)
 	if err != nil {
-		return "", err
+		return nil, "", err
 	}
 
 	client, err := c.newClientToken(ctx, user.AccessToken)
 	if err != nil {
-		return "", err
+		return nil, "", err
 	}
 
 	tag, _, err := client.GetTag(repo.Owner, repo.Name, tagName)
 	if err != nil {
-		return "", err
+		return nil, "", err
 	}
 
-	return tag.Commit.SHA, nil
+	commit, err := c.getCommitFromSHA(ctx, user, repo, tag.Commit.SHA)
```

Comment:
Might be cleaner to not include this call directly in here, but call it in the hook directly